Mathlib/Algebra/Module/CharacterModule.lean

```diff
@@ -0,0 +1,252 @@
+/-
+Copyright (c) 2023 Jujian Zhang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jujian Zhang, Junyan Xu
+-/
+
+import Mathlib.Algebra.Module.LinearMap
+import Mathlib.Algebra.Category.ModuleCat.Basic
+import Mathlib.Algebra.Category.GroupCat.Injective
+import Mathlib.Topology.Instances.AddCircle
+import Mathlib.Topology.Instances.Rat
+
+/-!
+# Character module of a module
+
+For commutative ring `R` and an `R`-module `M`, its character module `M⋆` is defined to be
+`ℤ`-linear maps `M ⟶ ℚ ⧸ ℤ`.
+
+`M⋆` also has an `R`-module structure given by `(r • f) m = f (r • m)`.
+
+## Main results
+
+- `CharacterModuleFunctor` : the contravariant functor of `R`-modules where `M ↦ M⋆` and
+an `R`-lineara map `l : M ⟶ N` induces an `R`-linear map `l⋆ : f ↦ l ∘ f` where `f : N⋆`.
+- `LinearMap.charaterfy_surjective_of_injective` : If `l` is injective then `l⋆` is surjective,
```

Comment:
what does charaterfy means? characterfy?

Reply:
To transform a linear map into a map between characters. For lack of a better name and wanting to use `LinearMap` dot notation, I choose this name

Reply:
OK, so maybe `characterify` would be better. (In any case, at least add a `c` before the `t`).

Reply:
done, the typo is slightly embracing 😅